x-pack/plugins/fleet/server/routes/agent_policy/handlers.ts

```diff
@@ -112,7 +112,7 @@ export const createAgentPolicyHandler: FleetRequestHandler<
 > = async (context, request, response) => {
   const coreContext = await context.core;
   const fleetContext = await context.fleet;
-  const soClient = fleetContext.epm.internalSoClient;
+  const soClient = appContextService.getInternalUserSOClient(request); // fleetContext.epm.internalSoClient;
```

Comment:
passing soClient here as the one used down here does not work for spaces:
https://github.com/elastic/kibana/blob/fec87b27a2359197e8f79b43118c88aaf79b8db5/x-pack/plugins/fleet/server/services/agent_policy_update.ts#L37-L40